### PR TITLE
Updated algos to include RS384, RS512

### PR DESCRIPTION
**Pull request by tiffanyachen**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs512.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    throw new ArgumentNullException( "digest" );
+
+                if ( digest.Length != SHA512_DIGEST_LENGTH )
+                    throw new ArgumentOutOfRangeException( "digest", String.Format("The digest must be {0} bytes for SHA-512", SHA512_DIGEST_LENGTH) );
+
+                if ( signature == null || signature.Length == 0 )
+                    throw new ArgumentNullException( "signature" );
+
+
+#if NET45
```
Comment (dsgouda) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs512.cs]: @shahabhijeet do we allow any other conditional variables than `FullNetFx` and `NETSTANDARD`?

Comment (schaabs) [reply src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs512.cs]: @dsgouda @shahabhijeet The build constants we are using here are defined in our project files, and we use these in several places throughout the KeyVault.Cryptography assembly to identify whether certain framework cryptography classes are available. Also the build constant FullNetFx didn't originally exist in our branch (my best guess is this was introduced after we branched).

I'd like to merge this change as is, and then after that review replacing all the build constants with the common ones at once, so it's consistent throughout all the cryptography classes. Thoughts?

**Comment by dsgouda** [comment]
@tiffanyachen Please join the azure org [here](aka.ms/azuregithub) to kick off the CI

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs256.cs [hunk 2]
```diff
@@ -50,8 +51,10 @@ public byte[] Sign( byte[] digest )
                 if ( digest == null || digest.Length == 0 )
                     throw new ArgumentNullException( "digest" );
 
-                if ( digest.Length != 32 )
-                    throw new ArgumentOutOfRangeException( "digest", "The digest must be 32 bytes for SHA-256" );
+                if ( digest.Length != SHA256_DIGEST_LENGTH
+     )
+                    throw new ArgumentOutOfRangeException( "digest", String.Format( "The digest must be {0} bytes for SHA-256", SHA256_DIGEST_LENGTH
+         ));
```
Comment (schaabs) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs256.cs]: Can you fix the formatting here?

File: src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs256.cs [hunk 3]
```diff
@@ -72,8 +75,10 @@ public bool Verify( byte[] digest, byte[] signature )
...
```
Comment (schaabs) [inline src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.Cryptography/Algorithms/Rs256.cs]: same as above

**Review by schaabs — CHANGES_REQUESTED** [review]
:clock1:

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by dsgouda** [comment]
@tiffanyachen Please fix the failing test cases

**Comment by dsgouda** [comment]
@tiffanyachen @schaabs gentle ping. Is this PR still being worked on?

**Comment by shahabhijeet** [comment]
@tiffanyachen please reopen this PR once you are ready to get this reviewed with the proposed changes.
Closing it for now.

**Pull request closed without merging**